app/src/test/java/com/pavesid/androidacademy/ui/MoviesViewModelTestMockito.kt

```diff
@@ -0,0 +1,64 @@
+package com.pavesid.androidacademy.ui
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.lifecycle.Observer
+import com.pavesid.androidacademy.MainCoroutineRule
+import com.pavesid.androidacademy.data.Movie
+import com.pavesid.androidacademy.repositories.MoviesRemoteRepositoryTest
+import com.pavesid.androidacademy.utils.Resource
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.Mockito.verify
+import org.mockito.junit.MockitoJUnitRunner
+
+@ExperimentalCoroutinesApi
+@RunWith(MockitoJUnitRunner::class)
+class MoviesViewModelTestMockito {
+
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    @get:Rule
+    var mainCoroutineRule = MainCoroutineRule()
+
+    private val repository = MoviesRemoteRepositoryTest()
+
+    private val coroutineDispatcher = TestCoroutineDispatcher()
```

Comment:
why not from DI?

Reply:
Because the repository is not from DI. And the repository is not from DI, because you need to be able to manually switch the response.